theseus/geometry/so3.py

```diff
@@ -3,24 +3,42 @@
 # This source code is licensed under the MIT license found in the
 # LICENSE file in the root directory of this source tree.
 
-from typing import Optional, cast
+from typing import Optional, Union, cast
 
 import torch
 
+import theseus.constants
+
 from .lie_group import LieGroup
+from .point_types import Point3
 
 
 class SO3(LieGroup):
     def __init__(
         self,
+        quaternion: Optional[torch.Tensor] = None,
         data: Optional[torch.Tensor] = None,
         name: Optional[str] = None,
         dtype: Optional[torch.dtype] = None,
     ):
-        raise NotImplementedError
+        if quaternion is not None and data is not None:
+            raise ValueError("Please provide only one of theta or data.")
```

Comment:
theta --> quaternion

Reply:
Thanks, Luis! This is very helpful.

Reply:
Hi Luis, [test_so3.py](https://github.com/facebookresearch/theseus/blob/taoshaf.add_so3_class/theseus/geometry/tests/test_so3.py) is added. Thanks!